tests/conftest.py

```diff
@@ -0,0 +1,27 @@
+import fnmatch
+import os
+
+import nengo
+from nengo.tests.conftest import *
+from nengo.utils.testing import find_modules, allclose, load_functions
+
+import nengo_spinnaker
+
+
+# A monkey patch hack to solve a problem in Nengo master
+def get_filename(self, ext=''):
+    modparts = self.module_name.split('.')[1:]
+    return "%s.%s.%s" % ('.'.join(modparts), self.function_name, ext)
+nengo.utils.testing.Recorder.get_filename = get_filename
+
+
+@pytest.fixture
+def Simulator():
+    return nengo_spinnaker.Simulator
```

Comment:
Would it be possible to monkeypatch/subclass this class before you return it, rather than modifying the simulator itself? You could also use the fixture's finaliser to call close, even if the test itself threw an exception and couldn't call close(). That may help the situation with the `InsufficientResourceError`s too.